client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +50,46 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final StreamDataRetention retentionConfig;
+
+    /**
+     * If a Stream's Retention Policy {@link RetentionPolicy} is set to
+     * 'CONSUMPTION' based retention {@link RetentionPolicy.RetentionType#CONSUMPTION} ,
+     * the Reader Group needs to periodically notify StreamCuts to Controller to indicate
+     * the point in the Stream till which it has completed consuming data.
+     * This can be done by setting the retentionConfig in ReaderGroupConfig {@link ReaderGroupConfig}
+     * to 'CONSUMPTION_BASED_USER_STREAMCUT' or 'CONSUMPTION_BASED_AT_LAST_CHECKPOINT'.
+     * If a Stream's Retention Policy is not Consumption based the ReaderGroupConfig should have retentionConfig = 'NO_IMPACT'.
+     *
+     * Note: It is incorrect to set ReaderGroup retentionConfig = 'NO_IMPACT' when
+     * a Streams' Retention Policy = 'CONSUMPTION' {@link RetentionPolicy.RetentionType#CONSUMPTION},
+     * because with NO_IMPACT the ReaderGroup will not notify consumption Stream-Cuts to Controller
+     * and so Consumption based retention of data in the Stream would not happen.
+     *
+     * NO_IMPACT - Read Positions of Readers do not impact Stream truncation/retention.
+     * CONSUMPTION_BASED_USER_STREAMCUT - User provides the StreamCut for truncation using Consumption Based Retention.
+     * CONSUMPTION_BASED_AT_LAST_CHECKPOINT - StreamCut corresponding to lastCompletedCheckpoint is auto-notified as truncation Stream-Cut for Consumption Based Retention.
```

Comment:
I think we can eliminate these separate bullets if we explain each in the paragraph above.

Reply:
Keeping these as they are not explained in the paragraph above.